### PR TITLE
Remove Bluebird

### PR DESCRIPTION
**Pull request by joshbwlng**
Remove Bluebird

Change-type: patch
Signed-off-by: Josh Bowling <josh@balena.io>

### CODE REVIEW

File: lib/events.ts
Comment (joshbwlng) [inline lib/events.ts:218]: We have been unknowingly (?) returning `undefined` for a while now because of Bluebird. See `resolve(result || undefined);` below. Want to get a second opinion on this result type change before merging. @LucianBuzzo @Hades32

Comment (Hades32) [reply lib/events.ts:218]: @joshbwlng is that really the case? I'm not familiar with `Stream`, but I had expected that the `close` event would only fire when it's actively closed (and that we'd get an `error` event when the server closes the connection). If that's the case, shouldn't we then always get a result?

And if that's not the case, shouldn't we `reject` if the result hasn't been defined on closing?

Comment (joshbwlng) [reply lib/events.ts:218]: @Hades32 You may be right in that we were never actually returning `undefined`, I didn't dig that deep. It could be a case in which using Bluebird allowed for improper values in its `resolve()` and we just never removed the `|| undefined` bit.

Comment (joshbwlng) [reply lib/events.ts]: Pushed another commit that removes `undefined` from possible return types in favor of `ExecuteContract` alone.

**Review by Hades32 — APPROVED** [review]

**Pull request merged**
